File: src/uva/Leetcode00002.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
public class Solution {
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
        ListNode head = null;
        ListNode node = null;
        ListNode l = l1,r = l2;
        int carry = 0;
        if (l != null||r != null) {
            int sum = sum(l,r,carry);
            carry = sum/10;
            sum = sum%10;
            head = node = new ListNode(sum);
            l = (l!=null?l.next:null);
            r = (r!=null?r.next:null);
            while (l!=null||r!=null||carry!=0) {
                sum = sum(l,r,carry);
                carry = sum/10;
                sum = sum%10;
                node.next = new ListNode(sum);
                node = node.next;
                l = (l!=null?l.next:null);
                r = (r!=null?r.next:null);
            }
        }
        return head;
    }
    public int sum(ListNode l, ListNode r, int carry) {
        int sum  = 0;
        if (l != null) {
            sum += l.val;
        }
        if (r != null) {
            sum += r.val;
        }
        sum += carry;
        return sum;
    }
}

File: src/uva/Leetcode00002.swift
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public var val: Int
 *     public var next: ListNode?
 *     public init(_ val: Int) {
 *         self.val = val
 *         self.next = nil
 *     }
 * }
 */
class Solution {
    func addTwoNumbers(l1: ListNode?, _ l2: ListNode?) -> ListNode? {
        var l : ListNode? = l1, r : ListNode? = l2, c = 0
        let head = ListNode(0)
        var node : ListNode = head;
        var vv = l != nil ? 1 : 0
        while l != nil || r != nil || c > 0 {
            var s : Int = (l?.val ?? 0) + (r?.val ?? 0) + c
            c = s / 10
            s = s % 10
            node.next = ListNode(s)
            node = node.next!
            l = l?.next
            r = r?.next
        }
        return head.next
    }
}
